File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonRed.java
package org.firstinspires.ftc.teamcode;

import com.bylazar.configurables.annotations.Configurable;
import com.pedropathing.geometry.Pose;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit;
import org.firstinspires.ftc.robotcore.external.navigation.Pose2D;

@Configurable
@TeleOp(name = "RED auton")
public class AutonRed extends AutonBase {
    //public static Config c = new Config(); // TODO: add definitions for all the values
    //THIS C IS INCORRECT, this is the values for AUTONRED, this needs to be changed
    public static Config c = new AutonBase.Config();
    @Override
    Config getConfig() {
        return c;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutonRed.kt
package org.firstinspires.ftc.teamcode

import com.pedropathing.geometry.Pose
import com.qualcomm.robotcore.eventloop.opmode.Autonomous

@Autonomous(name = "RED Auton")
class AutonRed : Auton() {
    override fun getWaypoints(): Waypoints {
        return w
    }

    companion object {
        @JvmField
        var w =
            Waypoints(
                Pose(56.0, 8.0, 90.0),
                Pose(40.0, 36.0, 180.0),
                Pose(0.0, 24.0, 0.0),
                Pose(-18.0, 0.0, 0.0),
                Pose(60.0, 96.0, 30.0)
            )
    }
}
